### PR TITLE
Add borrower inflator modifier, rename payBack function to repay

### PR DESCRIPTION
**Pull request by grandizzy**

### CODE REVIEW

File: src/PriceBuckets.sol [hunk 1]
```diff
@@ -142,6 +174,45 @@ contract PriceBuckets is IPriceBuckets {
...
+    function repay(
+        uint256 _amount,
+        uint256 _lup,
+        uint256 _inflator
+    ) public returns (uint256, uint256) {
+        Bucket storage curLup = buckets[_lup];
+
+        while (true) {
+            // accumulate bucket interest
+            if (curLup.debt > 0) {
```
Comment (EdNoepel) [inline src/PriceBuckets.sol:187]: Although `_amount` is getting decremented, it seems the `while` condition is looking at `curLup.debt`.  Shouldn't it `break` once `_amount == 0`?

Comment (EdNoepel) [reply src/PriceBuckets.sol:187]: NVM, I see that happens on line 201.

File: src/ERC20Pool.sol [hunk 2]
```diff
@@ -84,7 +82,38 @@ contract ERC20Pool is IPool {
         _buckets = new PriceBuckets();
     }
 
-    function addQuoteToken(uint256 _amount, uint256 _price) external {
+    modifier updateInflator() {
```
Comment (MikeHathaway) [inline src/ERC20Pool.sol]: Is there a reason for updating state in modifiers vs using an internal function? Was under the impression that the were typically used for authentication

Comment (grandizzy) [reply src/ERC20Pool.sol]: No reason, just a wrong assumption re modifiers usage on my side. I changed that, is also better from gas cost perspective as the borrower is loaded from storage only once now

Comment (MikeHathaway) [reply src/ERC20Pool.sol]: Gotcha, sounds good!

**Review by Exef — APPROVED** [review]

**Pull request merged**